raft-tech/TANF-app
# Fix/1744 permission guards

## Summary of Changes
Pull request closes #1744

Adds a `PermissionGuard` component for conditionally rendering `children` based on the user permissions stored in redux. Implemented for the "Data Files" tab and route to resolve 1744.

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && npm run start
```

1. Open http://localhost:3000/ and sign in.
1. In the Django admin, give your user the following settings
   * `account_approval_status`: `Approved`
   * `role`: `None`
1. In the frontend, observe that there is no "Data Files" navigation item. Attempting to navigate to `/data-files` will redirect you back to the home page.
1. In the Django admin, give your user the following settings.
   * `role`: `Data Analyst`
1. In the frontend, observe that you have been granted the `add_datafile` and `view_datafile` permissions. Observe that the "Data Files" navigation item is now visible and clickable. The Data Files page can be navigated to and used.
1. In the Django admin, set the user's `account_approval_status` to "Pending". Notice that the Data Files tab is no longer visible and the page is no longer accessible.

You can read through the added test cases to see more fine-grained behavior. 

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Only approved users with permissions to submit data (i.e. OFA Admin, Data Analyst , and Developer (in non-prod envs only)) can see the data-files page
+ [x] Only approved users with permissions to submit data, can click search and submit data files buttons
+ [x] Only approved users with permissions to submit data, can access previously submitted data files for their STT
+ [x] Permissions are up to date for each user group.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested? **N/A**
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
